### ISSUE TITLE
Go specific topics

### ISSUE THREAD

**Issue by robphoenix** [exercism/go#882]
This is a continuation of the discussion from #868 about including Go specific topics.

@shaleh
> robot-name: This will also be the first task to use methods and receive by reference. None of this is reflected in the topics.
> error-handling: This is one of the first uses of defer and recover. Again, no Go specific topics there.

@tleen
> We are free to add our own topics, at the same time if something we think should be a topic and is not, I feel it should be proposed first in the TOPICS.txt file. Sometimes things don't get included, which should not preclude it from our track but we should give the problem-spec contributors a fair crack at thinking on the topic as a global before including it locally.

In formulating what this issue should address I had begun to conceive of having a repo specific `TOPICS.txt` to list Go specific topics. As I thought this through I realised that many topics would be duplicated from the problem-specifications list. It started to feel that his wouldn't add much to documenting the exercises and helping users to identify areas of the Go language to learn and understand.

I think the better place for acknowledging Go specific topics is in exercise documentation. If we mark an exercise as being the first time a Go concept is introduced or being a good example of taking advantage of a Go specific concept then we can mention this in the README, either in the hints.md file or description.md file. This then gives us the opportunity to suggest concepts that might be useful and link to helpful information.

So, something like *You might want to read up about using `defer`*, rather than labelling the exercise with the topic `defer`.

The concern with either approach is whether this constitutes as too much hand-holding and leading users down particular paths. Sometimes these things are better learnt by doing things the wrong way and then finding out yourself how and why to do it differently or more idiomatically. Here we start to rely more on the code review aspect of exercism, for these topics to come up as part of the conversations on the site.

**In summary** I think any Go specific topic suggestions are perhaps best given in exercise documentation, leaving topic lists for more generalised subject matter. I also think we should be careful not to *lead the horse to water*, as it were, when offering guidance.

**Comment by tleen** [comment]
I may agree with you. I don't really see yet in nextercism how the global topics and local topics will mesh. So we may want to `defer` creating any GO unique topics till we can see how it is working with nextercism. :smile:

Either way if it is something specific to go I am very much in favor of pointing users to relevant tutorials & documentation for the topic. I would think that is part of the educational mission of the track.

**Comment by shaleh** [comment]
I am totally ok with it ending up in something like hints.

Often a book, tutorial, or whatever will ask the reader to think about solving a problem using a different tactic. Something like "ok, now let's try this recursively" or "or let's write an OO version". Adding hints that while a brute approach may work the reader may want to try using the singleton pattern or this task is best written using methods so they should read up on passing by reference.

**Comment by robphoenix** [comment]
Grand, thanks both. I will close this as I don't see any need for any further discussion.

So the rule of thumb is, we can look at adding further topics to the main `problem-specifications` repo topics list, otherwise we can defer to including any topics hints in the exercise README.
